### DISCUSSION TITLE
Number of people

### DISCUSSION THREAD

**Discussion by rutschi100** [mealie-recipes/mealie#2613] [Ideas]
Hi. First of all, huge praise for the UI, looks really great.

I am currently testing the app to see if it is something for me. Something caught my eye that I would like to discuss.

While I saw that after creating a recipe, I have no way to reduce the portion lengths while cooking. Here the example of 4 servings, is automatically considered a minimum, which makes no sense in my eyes. I live alone and have the recipe possibly from somewhere where you cook for several people.

![image](https://github.com/mealie-recipes/mealie/assets/50636192/a88dd179-e9ca-488c-9732-3406d073f30a)

Furthermore, I saw that if I add the recipes to a shopping list, I can increase the number of recipes. This does not make sense to me. Why would I want to cook 2x a recipe for 4 people. Wouldn't it be better to take the number of people instead of the number of recipes? I hope you understand what I mean.

![image](https://github.com/mealie-recipes/mealie/assets/50636192/1b7a4642-a0d9-403d-8637-e6600e09bb09)
![image](https://github.com/mealie-recipes/mealie/assets/50636192/9def2e73-98b1-4de5-ac7f-9bfc75b9cb7b)

I am curious about the answer.
Have a nice day :D

**Comment by michael-genson** [comment]
You can edit the number of servings in the recipe to fit your use case (you can make it 4 people, 4 servings, 100 people, 1 person, etc). 4 is just the default

**Reply by rutschi100** [reply]
Yes, that's true, but it forces me to go into edit mode while cooking. I.e. I can either increase the portions, or I have to edit the recipe.

So to avoid both of the problems I listed, when creating/importing the recipe, you would always have to convert it so that the portions are always 1. Is a workaround, but is messy. What do you think?

**Reply by michael-genson** [reply]
I'm not sure I understand; you only need to edit the serving size once. Then you can scale up and down as needed whenever you're cooking

**Reply by rutschi100** [reply]
I think I may not have made myself clear. Currently I can only increase the portions, but not decrease them. If there is a recipe for 4 people, I suggest introducing the feature to reduce it to 2 people without having to edit the recipe to increase usability.

On the subject of shopping lists: When adding ingredients to a shopping list, I suggest having the ability to adjust the number of servings rather than the default amount of the recipe. Currently it is not apparent how many servings the ingredient list is for.

Setting each recipe to 1 serving works as a workaround, but you have to manually convert each recipe before you can enter it into the app, although dividing by code shouldn't be a problem.

I hope this clears up my suggestions a bit. Thanks for your time and effort!

**Reply by michael-genson** [reply]
Ahh I see, I misunderstood. I agree it should be easier to adjust the serving size, but you can decrease it without editing by clicking on the servings:
![image](https://github.com/mealie-recipes/mealie/assets/71845777/cd6e4466-b9a9-4584-a9c0-60c1924558ad)

Regarding the shopping list - when adding to your list, it respects serving size, so you can set the scale _before_ hitting "add to shopping list" and it will add that many of the recipe. I agree that it could be clearer. I also think adding the servings (vs just the recipe qty) is a great suggestion

**Reply by JvdMaat** [reply]
I sorta get where OP is coming from .. If the recipe is 4 servings, the plus makes it 8 (double). The minus really should allow it to go to 2 (half).
Though what it is really doing is 4 times plus (4, 8, 12, 16, etc). So the minus (programmatically) would go from 4 straight to 0.

So the answer is simply to click on the 4 Servings part in red as shown above and manually type the multiplier. (0.5 to half, 0.25 for quarter, etc)

**Reply by rutschi100** [reply]
Exactly. I understand well how to deal with the fact that it is correct. But I think for a normal user this will not be understandable.

To be honest, I don't understand why the scale exists at all? In my opinion, working with serving is much more intuitive. To get from serving 4 to 2 you just have to calculate (( count / 4) * 2). Instead, you first have to figure out what is meant by scale and play around with it until you understand it. 

We know that for the users every additional click is too much. 
Please don't get me wrong. I just want to give constructive feedback here and not badmouth anything.

**Reply by JvdMaat** [reply]
Ah, so you're saying why not allow people to enter a number for the number of servings instead of scale, and have Mealie figure out the rest?
So if a recipe is 4 servings, you want to enter 6, and have Mealie just do a 1.5 multiplier on everything?

**Reply by rutschi100** [reply]
Yes, exactly. Both upwards and downwards. I would say that any number can be chosen, regardless of what the default is when creating the recipe.

The example of 4 to 6 is a multiplier of 1.5, but can be solved more generically so it could work with anything. Here pseudo code for it:

````
If the number of serving changes
	LOOP for each ingredient
		new amount of ingredient = ( ( amount of ingredient per original recipe / serving per original recipe ) * new serving )
	End of LOOP
End of IF
````

What do you think about it? Would something like that make sense?
